kratos/python/add_sparse_matrices_to_python.cpp

```diff
@@ -0,0 +1,217 @@
+
+
+
+    //************************************************************************************************************
+    //************************************************************************************************************
+    py::class_<SystemVector<double,IndexType>, SystemVector<double,IndexType>::Pointer >(m, "SystemVector", py::buffer_protocol())
+    .def(py::init<IndexType>())
+    .def(py::init<IndexType, DataCommunicator&>())
+    .def(py::init<SparseGraph<IndexType>&>())
+    .def(py::init<SparseContiguousRowGraph<IndexType>&>())
+    .def("GetComm", &SystemVector<double,IndexType>::GetComm)
+    .def("pGetComm", &SystemVector<double,IndexType>::pGetComm)
+    .def("Size", &SystemVector<double,IndexType>::size)
+    .def("size", &SystemVector<double,IndexType>::size)
+    .def("Clear", &SystemVector<double,IndexType>::Clear)
+    .def("SetValue", &SystemVector<double,IndexType>::SetValue)
+    .def("__setitem__", [](SystemVector<double,IndexType>& self, const IndexType i, const double value){self[i] = value;} )
+    .def("__getitem__", [](const SystemVector<double,IndexType>& self, const IndexType i){return self[i];} )
+    .def("Add", &SystemVector<double,IndexType>::Add)
+    .def("BeginAssemble", &SystemVector<double,IndexType>::BeginAssemble)
+    .def("FinalizeAssemble", &SystemVector<double,IndexType>::FinalizeAssemble)
+    .def("BeginAssemble", &SystemVector<double,IndexType>::BeginAssemble)
+    .def("data", [](SystemVector<double,IndexType>& self){
+       return self.data();
+    })
+    .def_buffer([](SystemVector<double,IndexType>& self) -> py::buffer_info {
+        return py::buffer_info(
+            &(self.data())[0],                               /* Pointer to buffer */
+            sizeof(double),                          /* Size of one scalar */
+            py::format_descriptor<double>::format(), /* Python struct-style format descriptor */
+            1,                                      /* Number of dimensions */
+            { self.size() },                 /* Buffer dimensions */
+            { sizeof(double)  }             /* Strides (in bytes) for each index */
+        );
+    })
+    .def("Assemble", [](SystemVector<double,IndexType>& self, 
+                        Vector& values, 
+                        std::vector<IndexType>& indices){
+       self.Assemble(values,indices);
+    })
+
+    //inplace
+    .def("__iadd__", [](SystemVector<double,IndexType>& self, const SystemVector<double,IndexType>& other_vec){self += other_vec; }, py::is_operator())
+    .def("__isub__", [](SystemVector<double,IndexType>& self, const SystemVector<double,IndexType>& other_vec){self -= other_vec;  }, py::is_operator())
+    .def("__imul__", [](SystemVector<double,IndexType>& self, const double& value){ self*=value; }, py::is_operator())
+    .def("__itruediv__", [](SystemVector<double,IndexType>& self, const double& value){ self/=value; }, py::is_operator())
+
+    //out of place
```

Comment:
Clean unused?

Reply:
i would like to add the ones that are commented, however they need move semantics for ublas vectors (see #9541 )